lib/libmuser.c

```diff
@@ -77,16 +77,21 @@ typedef struct {
     _Static_assert(sizeof(s) - offsetof(s, m) == sizeof(t), \
         #t " " #m " must be last member in " #s)
 
+struct pci {
+    uint32_t                irq_count[LM_DEV_NUM_IRQS];
+    lm_reg_info_t           reg_info[LM_DEV_NUM_REGS];
+    lm_pci_config_space_t   *config_space;
+    struct caps             *caps;
+};
+
 struct lm_ctx {
     void                    *pvt;
     dma_controller_t        *dma;
     int                     fd;
     int (*reset)            (void *pvt);
     lm_log_lvl_t            log_lvl;
     lm_log_fn_t             *log;
-    lm_pci_info_t           pci_info;
-    lm_pci_config_space_t   *pci_config_space;
-    struct caps             *caps;
+    struct pci              pci;
```

Comment:
Do we want to put this in a union already, or leave it for later?

Reply:
Let's leave it for later, it makes no difference.